contrib/bdii/htcondor-ce-provider

```diff
@@ -128,21 +131,24 @@ def main():
     topologies = {}
     for ad in coll.query(htcondor.AdTypes.Startd, 'State=!="Owner"', ['Arch', 'OpSys', 'OpSysMajorVer', 'OpSysName',
                                                                       'DetectedCpus', 'DetectedMemory', 'Machine']):
-        if ad['Machine'] not in total_cores:
-            total_cores[ad['Machine']] = ad['DetectedCpus']
-            if ad['Arch'] not in topologies:
-                topologies[ad['Arch']] = {}
-            if ad['OpSys'] not in topologies[ad['Arch']]:
-                topologies[ad['Arch']][ad['OpSys']] = {}
-            if ad['OpSysName'] not in topologies[ad['Arch']][ad['OpSys']]:
-                topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']] = {}
-            if ad['OpSysMajorVer'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']]:
-                topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']] = {}
-            if ad['DetectedCpus'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']]:
-                topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']] = {}
-            if ad['DetectedMemory'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']]:
-                topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']][ad['DetectedMemory']] = 0
-            topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']][ad['DetectedMemory']] += 1
+        try:
+            if ad['Machine'] not in total_cores:
+                total_cores[ad['Machine']] = ad['DetectedCpus']
+                if ad['Arch'] not in topologies:
+                    topologies[ad['Arch']] = {}
+                if ad['OpSys'] not in topologies[ad['Arch']]:
+                    topologies[ad['Arch']][ad['OpSys']] = {}
+                if ad['OpSysName'] not in topologies[ad['Arch']][ad['OpSys']]:
+                    topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']] = {}
+                if ad['OpSysMajorVer'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']]:
+                    topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']] = {}
+                if ad['DetectedCpus'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']]:
+                    topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']] = {}
+                if ad['DetectedMemory'] not in topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']]:
+                    topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']][ad['DetectedMemory']] = 0
+                topologies[ad['Arch']][ad['OpSys']][ad['OpSysName']][ad['OpSysMajorVer']][ad['DetectedCpus']][ad['DetectedMemory']] += 1
+        except KeyError:
+            pass
```

Comment:
I think this entire block can be simplified with a series of default dicts (I can submit a PR). Do you know which key actually had issues? Catching all `KeyErrors` here seems like a hammer to solve a potential logic problem

Reply:
Actually I think there's better exception handling that we can do here, too. This is fine as is but I'll submit another PR to address it

Reply:
The error was with OpSysMajorVer but any attribute could be missing from a malformed ad.